components/ai_chat/core/browser/ai_chat_database.cc

```diff
@@ -154,7 +154,9 @@ AIChatDatabase::AIChatDatabase(const base::FilePath& db_file_path,
       db_(sql::DatabaseOptions().set_page_size(4096).set_cache_size(1000),
           sql::Database::Tag("AIChatDatabase")),
       encryptor_(std::move(encryptor)) {
-  base::AssertLongCPUWorkAllowed();
+  // Constructor is the only thing in this class which is not on the same
+  // sequence, because it is called on the original thread.
+  DETACH_FROM_SEQUENCE(sequence_checker_);
```

Comment:
I wonder why I don't see this pattern on a recent Chromium example? https://source.chromium.org/chromium/chromium/src/+/main:components/history_embeddings/sql_database.cc;l=93. Please get someone more familiar with this to code review this part.

Reply:
Underlining sql/database.cc actually has this: https://source.chromium.org/chromium/chromium/src/+/main:sql/database.cc;l=371-373;drc=4e673771b1ee61f0e9f854e2d1420f353c67c401;bpv=1;bpt=1
Being able to construct the object on the original sequence makes it possible to mock the database in tests, all access after this is still guarded by the sequence checker.
I had asked @darkdh to review the backend code too.

Reply:
yes, this is legit when constructing it in different sequence and then later pass it into a sequence bound.